blocks/serialization.py

```diff
@@ -577,18 +593,35 @@ def __call__(self, id_):
         # avoid creating the same object more than once.
         if id_ not in self._cache:
             components = _unmangle_parameter_name(id_)
-            self._cache[id_] = components[0](self.parameters[components[1]])
+            self._cache[id_] = components[0](
+                components[1], self.parameters[components[2]])
         return self._cache[id_]
 
 
-def _mangle_parameter_name(type_, name):
-    return '#{}.{}'.format(_ARRAY_TYPE_MAP[type_], name)
+def _mangle_parameter_name(parameter, name):
+    array_type = type(parameter.container.storage[0])
+    context_name = (parameter.context_name
+                    if pygpu and
+                    isinstance(parameter, pygpu.gpuarray.GpuArray)
+                    else None)
+    if isinstance(context_name, str) and '.' in context_name:
+        raise ValueError("context name must not contain dots")
+    return '#1{}.{}.{}'.format(
+        _ARRAY_TYPE_MAP[array_type], context_name, name)
```

Comment:
As I told, my comment was for safety. I think it is a good habit to not change input order. You don't know if someone else have code outside that reuse it. But as this is function marked as internal, I'm fine to merge this now.

I let you merge as I don't know if you consider blocking that the coverage went down.